### ISSUE TITLE
Error implementation of deeplabv3plus

### ISSUE THREAD

**Issue by jianlong-yuan** [open-mmlab/mmsegmentation#1465]
We found that the method you implement is different from the official implementation. For example, the channel in decoder is different from the official implementation. Why don't you use same architecture ?

**Comment by MengzhangLI** [comment]
Please list detailed differences, thank you.

**Comment by MengzhangLI** [comment]
Due to its old age, please list detailed differences which are not identical with original paper. Thus we could try to answer this question.

**Comment by jianlong-yuan** [comment]
1. Channel in decode_head' channels=512, acturally, deeplabv3+' channel is 256.
2. Deeplabv3+ doesn't have aux layers.
model = dict(
    decode_head=dict(
        type='DepthwiseSeparableASPPHead',
        in_channels=2048,
        in_index=3,
        channels=512,
        dilations=(1, 12, 24, 36),
        c1_in_channels=256,
        c1_channels=48,
        dropout_ratio=0.1,
        num_classes=19,
        norm_cfg=norm_cfg,
        align_corners=False,
        loss_decode=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0)),)

**Comment by MengzhangLI** [comment]
Thanks for your feedback.

I've checked DeeplabV3+ of mmseg, it is pushed along with initial commit:
https://github.com/open-mmlab/mmsegmentation/commit/b2724da80bf2dc203b3db69497541ad52ce74ba0.

If it is really different from original paper, we would add notification in Deeplabv3+ readme to remind users asap.

Best,

**Comment by jianlong-yuan** [comment]
any news?

**Comment by MengzhangLI** [comment]
> any news?

Sorry for late reply. That because when re-implementing Deeplabv3plus, it has been a long time and we just followed some popular/ famous third-party repo of Deeplabv3plus, which are also widely used. Thus we do not strictly follow setting of original paper.

**Comment by jianlong-yuan** [comment]
some popular/ famous third-party repos ? Where is? Could you provide it? Let's learn about it?

**Comment by timothylimyl** [comment]
@MengzhangLI

I agree with @jianlong-yuan , the decode_head channels (if following original) is supposed to be 256, as per the ![paper (page6, Proposed Decoder](https://arxiv.org/pdf/1802.02611.pdf):

"We apply another 1 × 1 convolution on the
low-level features to reduce the number of channels, since the corresponding low-
level features usually contain a large number of channels (e.g., 256 or 512) which
may outweigh the **importance of the rich encoder features (only 256 channels in
our model)** and make the training harder"

original code ref: ![1](https://github.com/tensorflow/models/blob/231c6bd08f40ff01be99b717e2619fb18c05c3c2/research/deeplab/common.py#L133) ![2](https://github.com/tensorflow/models/blob/231c6bd08f40ff01be99b717e2619fb18c05c3c2/research/deeplab/model.py#L450), just took a quick look, the code is very messy but I think these are the references to the aspp filters

However, I think this is just a minor hyperparameter tweak.  Regarding the aux layers, I think it can be commented out if you do not want to use it during training (have not personally tried it).
